Bluejay.asm

```diff
@@ -3924,6 +4060,7 @@ beep_delay_set:
 wait_for_start_no_beep:
 	jb	Flag_Telemetry_Pending, wait_for_start_check_rcp
 	call	dshot_tlm_create_packet		; Create telemetry packet (0 rpm)
+	call 	scheduler_run
```

Comment:
Should this happen after the beep routine is done? because during beep routines we want no interrupts.. and if scheduler starts a timer which causes an interrupt, then the registers used by beep routine may get overwritten

Reply:
this call is inserted here to allow sending extended telemetry while waiting for the initial rc pulse value, for example in Betaflight motors tab, so you don't need to spin the motors to start seeing temperature values. This was done to replicate Blheli_32 behaviour.

Reply:
@damosvil I am not sure if this answers the question - will there be interrupts because of that while the beep routine is running?

Reply:
Np, there are no interrupts resulting of scheduler_run call